volume-providers/vpc/provider/provider.go

```diff
@@ -62,6 +62,31 @@ func NewProvider(conf *config.Config, logger *zap.Logger) (local.Provider, error
 	if conf.Bluemix == nil || conf.VPC == nil {
 		return nil, errors.New("Incomplete config for VPCBlockProvider")
 	}
+
+	//Do config validation and enable only one generationType (i.e VPC-Classic | VPC-NG)
+	gcConfigFound := (conf.VPC.EndpointURL != "") && (conf.VPC.TokenExchangeURL != "") && (conf.VPC.APIKey != "") && (conf.VPC.ResourceGroupID != "")
+	g2ConfigFound := (conf.VPC.G2EndpointURL != "") && (conf.VPC.G2TokenExchangeURL != "") && (conf.VPC.G2APIKey != "") && (conf.VPC.G2ResourceGroupID != "")
+	//if both config found, look for VPCTypeEnabled, otherwise default to GC
+	//Incase of NG configurations, override the base properties.
+	if (gcConfigFound && g2ConfigFound && conf.VPC.VPCTypeEnabled == "g2") || (!gcConfigFound && g2ConfigFound) {
+
+		conf.VPC.EndpointURL = conf.VPC.G2EndpointURL
+		conf.VPC.TokenExchangeURL = conf.VPC.G2TokenExchangeURL
+		conf.VPC.APIKey = conf.VPC.G2APIKey
+		conf.VPC.ResourceGroupID = conf.VPC.G2ResourceGroupID
+
+		//Set API Generation As 2 (if unspecified in config/ENV-VAR)
+		if conf.VPC.G2VPCAPIGeneration <= 0 {
```

Comment:
as of now there are two possible generation values 1 and 2, we can check `conf.VPC.G2VPCAPIGeneration != 2` then set 2

Reply:
As these values are passed to the REST calls  (generation and VERSION, Lets keep it in configuration file ,  this can be optional/auto-filled-when-not-provided

Reply:
This has been kept for configuration purpose through ENV or Secret (incase the integer value changes). if its UNSET, then it will default to "2" . External lib consumer should ideally set in their config toml. We made it optional for us.

conf.VPC.G2VPCAPIGeneration != 2 then set 2   is equivalent to 
conf.VPC.G2VPCAPIGeneration = 2  (no benefit of doing a "!=" check)

Lets keep a provision for getting the 'generation' and 'version' from config file (whatever user sets)